api/http/server.go

```diff
@@ -865,6 +875,57 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *http.Request) {
 	http.ServeContent(w, r, fileName, time.Now(), newBufferedReadSeeker(reader, getFileBufferSize))
 }
 
+// HandleGetTag responds to the following request
+//    - bzz-tag:/<manifest>  and
+//    - bzz-tag:/?tagId=<tagId>
+// Clients should use root hash or the tagID to get the tag counters
+func (s *Server) HandleGetTag(w http.ResponseWriter, r *http.Request) {
+	getTagCount.Inc(1)
+	uri := GetURI(r.Context())
+	fileAddr := uri.Address()
+
+	var tag *chunk.Tag
+	if fileAddr == nil {
+		tagString := r.URL.Query().Get("tagId")
+		if tagString == "" {
+			getTagFail.Inc(1)
+			respondError(w, r, "Missing one of the mandatory argument", http.StatusBadRequest)
+		}
+
+		u64, err := strconv.ParseUint(tagString, 10, 32)
+		if err != nil {
+			getTagFail.Inc(1)
+			respondError(w, r, "Invalid tagId argument", http.StatusBadRequest)
+		}
+		tagId := uint32(u64)
+
+		tag, err = s.api.Tags.Get(tagId)
+		if err != nil {
+			getTagNotFound.Inc(1)
+			respondError(w, r, "Tag not found", http.StatusNotFound)
+		}
+	} else {
+
+		tagByFile, err := s.api.Tags.GetByAddress(fileAddr)
+		if err != nil {
+			getTagNotFound.Inc(1)
+			respondError(w, r, "Tag not found", http.StatusNotFound)
+		}
+		tag = tagByFile
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Cache-Control", "no-cache, private, max-age=0")
+	r.Header.Del("ETag")
+	w.WriteHeader(http.StatusOK)
+	bytesToSend, err := tag.MarshalBinary()
```

Comment:
`MarshalBinary` gives back a byte slice.
use `json.Marshal()` instead

Reply:
This will need that i export all the fields of Tag. That means i change many files. Is it okay?

Reply:
not necessary. just have `Tag` implement `MarshalJSON`/`UnmarshalJSON` ([link](https://play.golang.org/p/kN-_1lLoggl))

Reply:
Also, MarshalBinary returns the binary representation fo tag, but the json content type header is sent. I think that sending json would be more appropriate.

Reply:
Now the API returns stuff in json encoded format.